models/surveys.go

```diff
@@ -276,6 +300,131 @@ func (api *API) PostSurveyDetails(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// PostSurveyClassifiers endpont handler - creates a new survey classifier
+func (api *API) PostSurveyClassifiers(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	surveyID := vars["surveyId"]
+
+	// Check survey exists and get it's PK
+	var surveyPK int
+	err := api.GetSurveyPKByID.QueryRow(surveyID).Scan(&surveyPK)
+
+	if err == sql.ErrNoRows {
+		re := NewRESTError("404", "Survey not found")
+		data, err := json.Marshal(re)
+		if err != nil {
+			http.Error(w, "Error marshalling NewRestError JSON", http.StatusInternalServerError)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+		w.WriteHeader(http.StatusNotFound)
+		w.Write(data)
+
+		return
+	}
+
+	if err != nil {
+		http.Error(w, "Error creating classifier type selector for survey '"+surveyID+"' - "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error creating classifier type selector for survey '"+surveyID+"' - "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	var postData []ClassifierTypeSelector
+	err = json.Unmarshal(body, &postData)
+	if err != nil {
+		http.Error(w, "Error unmarshalling JSON", http.StatusBadRequest)
+		return
+	}
+
+	// Start database transaction
+	tx, err := api.DB.Begin()
+	if err != nil {
+		http.Error(w, "Error creating classifier type selector for survey '"+surveyID+"' - "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Prepare statements in transaction
+	txCreateSurveyClassifierTypeSelectorStmt := tx.Stmt(api.CreateSurveyClassifierTypeSelectorStmt)
+	txCreateSurveyClassifierTypeStmt := tx.Stmt(api.CreateSurveyClassifierTypeStmt)
+	txGetClassifierTypeSelectorStmt := tx.Stmt(api.GetClassifierTypeSelectorStmt)
+
+	// Initialise response object
+	createdClassifiers := make([]ClassifierTypeSelector, len(postData))
+
+	for classifierIndex, classifierTypeSelector := range postData {
```

Comment:
Why is the API designed to create multiple survey classifiers. The API would be nicer and implementation simpler if it only handled creating one and the client is responsible for calling multiple times